src/chain.h

```diff
@@ -613,6 +628,12 @@ class CDiskBlockIndex : public CBlockIndex
         READWRITE(vMintDenominationsInBlock);
         READWRITE(fProofOfFullNode);
 
+        if (fCheckVeilDataHash && nTime >= nPowTimeStampActive) {
```

Comment:
I worry about using nTime instead of a block height. May cause issues? One thing that we are for sure about is that height is consistently incrementing by 1. Time can be give or take. And I believe that our PoS intentionally adds variation to the time when I went through it. Could cause valid blocks to fail?

Reply:
No this wont allow for valid block to fail. Time is always changing between blocks. We have a max time drift allowed. So about after 10 minutes after the time that we choose to activate. All blocks must be in ahead of activation time. At time point we will know that all blocks going forward will be mined using the new 3 Pow Algos.